bentoml/_internal/frameworks/pickle.py

```diff
@@ -0,0 +1,281 @@
+        name: str,
+        resource_quota: t.Optional[t.Dict[str, t.Any]],
+        batch_options: t.Optional[t.Dict[str, t.Any]],
+        model_store: "ModelStore" = Provide[BentoMLContainer.model_store],
+    ):
+        super().__init__(name, resource_quota, batch_options)
+        model_info, model_file = _get_model_info(tag, model_store)
+        self._model_store = model_store
+        self._model_info = model_info
+        self._model_file = model_file
+        self._function_name = function_name
+
+    @property
+    def num_replica(self) -> int:
+        return 1
+
+    @property
+    def required_models(self) -> t.List[Tag]:
+        return [self._model_info.tag]
+
+    # pylint: disable=attribute-defined-outside-init
+    def _setup(self) -> None:
+        with open(self._model_file, "rb") as f:
+            self._model = pickle.load(f)
+        self._infer_func = getattr(self._model, self._function_name)
+
+    def _run(  # type: ignore[reportIncompatibleMethodOverride]
+        self,
+        inputs: t.Union["ext.NpNDArray", "ext.PdDataFrame"],
+    ) -> "ext.NpNDArray":
+        return self._infer_func(inputs)
+
+
+@inject
+def load_runner(
+    tag: t.Union[str, Tag],
+    function_name: str = "predict",
+    *,
+    name: t.Optional[str] = None,
+    resource_quota: t.Union[None, t.Dict[str, t.Any]] = None,
+    batch_options: t.Union[None, t.Dict[str, t.Any]] = None,
+    model_store: "ModelStore" = Provide[BentoMLContainer.model_store],
+) -> "BaseRunner":
+    """
+    Runner represents a unit of serving logic that can be scaled horizontally to
+    maximize throughput. :func:`bentoml.pickle.load_runner` implements a Runner class that
+    wraps the commands that dump and load a pickled object, which optimizes it for the BentoML runtime.
+
+    Args:
+        tag (:code:`Union[str, Tag]`):
+            Tag of a saved model in BentoML local modelstore..
+        function_name (:code:`str`, `optional`, default to :code:`predict`):
+            Predict function used by a given pickled model.
+        resource_quota (:code:`Dict[str, Any]`, default to :code:`None`):
+            Dictionary to configure resources allocation for runner.
+        batch_options (:code:`Dict[str, Any]`, default to :code:`None`):
+            Dictionary to configure batch options for runner in a service context.
+        model_store (:mod:`~bentoml._internal.models.store.ModelStore`, default to :mod:`BentoMLContainer.model_store`):
+            BentoML modelstore, provided by DI Container.
+
+    Returns:
+        :obj:`~bentoml._internal.runner.Runner`: Runner instances for the target :mod:`bentoml.pickle` model
+
+    Examples:
+
+    .. code-block:: python
+
+        import bentoml
+
+        runner = bentoml.pickle.load_runner("my_model:latest")
+        runner.run([[1,2,3,4]])
+    """  # noqa
+    tag = Tag.from_taglike(tag)
+    if name is None:
+        name = tag.name
+
+    if batch_options and batch_options.get("enabled"):
```

Comment:
batch_options is only applicable for `Runner`, in case user wants to turn off the micro-batching behavior in Runner that supports processing batch input. Runner implementation should not use the `batch_option`.

In this case it should allow user to declare if the function/object being pickled supports processing a batch input. e.g.:

```
obj = train_model()
bentoml.picklable_model.save(obj, batch=True, method="__call__")
```

The save method should save the `batch=True` in the model options field, which can be used during `load_runner`, to decide if it should return a Runner or SimpleRunner